### PR TITLE
Framework: Update lib/inflight and clients

### PR DESCRIPTION
**Pull request by blowery**
Expose more of the lifecycle of `lib/inflight`, add tests, update callers.

To test, run the tests (CI will do this for you) and use the Reader.

### CODE REVIEW

File: client/lib/feed-lookup/index.js [hunk 1]
```diff
@@ -1,42 +1,43 @@
...
+import { isEmpty, head } from 'lodash';
+import LruCache from 'lru';
+
+/**
+ * Internal Dependencies
+ */
+import { isRequestInflight } from 'lib/inflight';
+import wpcom from 'lib/wp';
 
 const cache = new LruCache( 10 );
```
Comment (samouri) [inline client/lib/feed-lookup/index.js]: is 10 too small?

Comment (blowery) [inline client/lib/feed-lookup/index.js]: no, this practically never gets used

Comment (blowery) [inline client/lib/feed-lookup/index.js]: afaict this is only used by the feed posts controller to lookup urls like `/read/feeds/mgoblog.com/posts` and redirect them to the relevant feed id

Comment (blowery) [inline client/lib/feed-lookup/index.js]: which is probably why the broken usage of `inflight` never really mattered.

File: client/lib/inflight/test/index.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import {
+	isRequestInflight,
+	markRequestInflight,
+	completeRequest,
+	trackPromise,
+	requestTracker,
+	_clear,
+} from '../';
+
+describe( 'inflight', () => {
```
Comment (samouri) [inline client/lib/inflight/test/index.js]: beautiful tests

File: client/lib/inflight/test/index.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+				() => expect( false ).to.be.true,
+				() => expect( isRequestInflight( key ) ).to.be.false
+			);
+		} );
+	} );
+	context( 'requestTracker', () => {
+		it( 'should track a good request', ( done ) => {
+			const cb = ( err ) => {
+				if ( err ) {
+					expect( true ).to.be.false;
```
Comment (samouri) [inline client/lib/inflight/test/index.js]: whats the point of this if?

**Comment by samouri** [comment]
Code LGTM (only had a minor question).
Testing now

**Comment by samouri** [comment]
Everything LGTM when i test out the branch! 🚢

File: client/lib/inflight/test/index.js (hunk 3 above)
Comment (blowery) [inline client/lib/inflight/test/index.js]: ah, we don't expect `err` to be true. it's testing that `requestTracker` is correctly passing through a falsey value for err. Can simplify.

**Comment by blowery** [comment]
@samouri  cleaned up tests in 901e62be3

**Comment by samouri** [comment]
:shipit:

**Pull request merged**